be/src/vec/exec/vjdbc_connector.cpp

```diff
@@ -0,0 +1,361 @@
+Status JdbcConnector::open() {
+    if (_is_open) {
+        LOG(INFO) << "this scanner of jdbc already opened";
+        return Status::OK();
+    }
+
+    JNIEnv* env = nullptr;
+    RETURN_IF_ERROR(JniUtil::GetJNIEnv(&env));
+    RETURN_IF_ERROR(JniUtil::GetGlobalClassRef(env, JDBC_EXECUTOR_CLASS, &_executor_clazz));
+    RETURN_IF_ERROR(JniUtil::GetGlobalClassRef(env, "java/util/List", &_executor_list_clazz));
+    RETURN_IF_ERROR(JniUtil::GetGlobalClassRef(env, "java/lang/Object", &_executor_object_clazz));
+    RETURN_IF_ERROR(JniUtil::GetGlobalClassRef(env, "java/lang/Boolean", &_executor_uint8_t_clazz));
+    RETURN_IF_ERROR(JniUtil::GetGlobalClassRef(env, "java/lang/Byte", &_executor_int8_t_clazz));
+    RETURN_IF_ERROR(JniUtil::GetGlobalClassRef(env, "java/lang/Short", &_executor_int16_t_clazz));
+    RETURN_IF_ERROR(JniUtil::GetGlobalClassRef(env, "java/lang/Integer", &_executor_int32_t_clazz));
+    RETURN_IF_ERROR(JniUtil::GetGlobalClassRef(env, "java/lang/Long", &_executor_int64_t_clazz));
+    RETURN_IF_ERROR(JniUtil::GetGlobalClassRef(env, "java/lang/Float", &_executor_float_clazz));
+    RETURN_IF_ERROR(JniUtil::GetGlobalClassRef(env, "java/lang/Float", &_executor_double_clazz));
+    RETURN_IF_ERROR(_register_func_id(env));
+
+    // Add a scoped cleanup jni reference object. This cleans up local refs made below.
+    JniLocalFrame jni_frame;
+    {
+        std::string local_location;
+        std::hash<std::string> hash_str;
+        auto function_cache = UserFunctionCache::instance();
+        RETURN_IF_ERROR(function_cache->get_jarpath(hash_str(_conn_param.resource_name),
+                                                    _conn_param.driver_path,
+                                                    _conn_param.driver_checksum, &local_location));
+
+        TJdbcExecutorCtorParams ctor_params;
+        ctor_params.__set_jar_location_path(local_location);
+        ctor_params.__set_jdbc_url(_conn_param.jdbc_url);
+        ctor_params.__set_jdbc_user(_conn_param.user);
+        ctor_params.__set_jdbc_password(_conn_param.passwd);
+        ctor_params.__set_jdbc_driver_class(_conn_param.driver_class);
+
+        jbyteArray ctor_params_bytes;
+        // Pushed frame will be popped when jni_frame goes out-of-scope.
+        RETURN_IF_ERROR(jni_frame.push(env));
+        RETURN_IF_ERROR(SerializeThriftMsg(env, &ctor_params, &ctor_params_bytes));
+        _executor_obj = env->NewObject(_executor_clazz, _executor_ctor_id, ctor_params_bytes);
+    }
+    RETURN_ERROR_IF_EXC(env);
+    RETURN_IF_ERROR(JniUtil::LocalToGlobalRef(env, _executor_obj, &_executor_obj));
+    _is_open = true;
+    return Status::OK();
+}
+
+Status JdbcConnector::query(const std::string& table, const std::vector<std::string>& fields,
+                            const std::vector<std::string>& filters, const int64_t limit) {
+    if (!_is_open) {
+        return Status::InternalError("Query before open of jdbc query.");
+    }
+
+    _sql_str = "SELECT ";
```

Comment:
the direct operator of string will cause perfermance problem, may should use fmt buffer
 here.